src/main/java/org/icgc_argo/workflow/relay/config/elastic/ElasticsearchStartupListener.java

```diff
@@ -39,62 +38,84 @@
 import org.springframework.core.io.Resource;
 import org.springframework.stereotype.Component;
 
-@Profile("index")
+import static org.icgc_argo.workflow.relay.util.StringUtilities.inputStreamToString;
+
+@Profile({"index", "graphlog"})
 @Slf4j
 @Component
 public class ElasticsearchStartupListener implements ApplicationListener<ContextRefreshedEvent> {
 
   private final RestHighLevelClient client;
   private final ElasticsearchProperties properties;
+  private final ProfileManager profileManager;
 
   @Value("classpath:run_log_mapping.json")
   private Resource workflowIndexMapping;
 
   @Value("classpath:task_log_mapping.json")
   private Resource taskIndexMapping;
 
+  @Value("classpath:workflow_graph_log_mapping.json")
+  private Resource graphLogIndexMapping;
+
   @Autowired
   public ElasticsearchStartupListener(
-      @NonNull RestHighLevelClient client, @NonNull ElasticsearchProperties properties) {
+      @NonNull RestHighLevelClient client,
+      @NonNull ElasticsearchProperties properties,
+      @NonNull ProfileManager profileManager) {
```

Comment:
Now that i see how you are using `ProfileManager` you might as well just autowire the `Environment` here and pull the list of profiles in this constructor. One less class in your codebase then with no loss in readability or expressiveness.